sdk/client/src/stream/streaming_client.rs

```diff
@@ -229,9 +252,32 @@ impl StreamingClient {
         self.subscriptions.write().await.remove(id);
     }
 
+    async fn clear_subscription_if_not_ok(&self, id: &Id) {
+        if let Some(sender) = self.subscriptions.read().await.get(&id) {
+            if !sender.message_received.load(Ordering::Relaxed) {
+                debug!("StreamingClient OkTimeout for id: {:?}", &id);
+                sender
+                    .sender
+                    .send(Err(StreamError::subscription_ok_timeout()))
+                    .await
+                    .ok();
+                self.subscriptions.write().await.remove(id);
+            }
+        }
+    }
+
     async fn get_and_register_id(&self) -> SubscriptionStreamResult {
         let id = self.client.read().await.get_next_id();
         let receiver = self.register_subscription(id.clone()).await?;
+
+        debug!("StreamingClient starting OkTimeout task for id: {:?}", &id);
+        let clone = self.clone();
+        let id2 = id.clone();
+        tokio::task::spawn(async move {
+            sleep(Duration::from_millis(clone.config.ok_timeout_millis)).await;
+            clone.clear_subscription_if_not_ok(&id2).await;
+        });
```

Comment:
This could probably be done much more efficiently, without the sleep and without the need for an atomic bool.

The idea is that when you register a new subscription you want to have some sort of timeout such that if you don't see an `Ok` from the server you want to drop it and return an error to the user. This could probably be done sort of like this (using pseudocode):

```
fn subscribe(&self, ...) -> Result<Subscription> {
    let subscription = self.new_subscription(...)?;
    timeout(60, subscription.wait_for_ok())?;
    Ok(subscription)
}
```